cmake/sysbuild/image_signing.cmake

```diff
@@ -74,6 +74,17 @@ function(zephyr_mcuboot_tasks)
     return()
   endif()
 
+  # Fetch devicetree details for flash and slot information
+  dt_chosen(flash_node PROPERTY "zephyr,flash")
+  dt_nodelabel(slot0_flash NODELABEL "slot0_partition" REQUIRED)
+  dt_prop(slot_size PATH "${slot0_flash}" PROPERTY "reg" INDEX 1 REQUIRED)
+  dt_prop(write_block_size PATH "${flash_node}" PROPERTY "write-block-size")
+
+  if(NOT write_block_size)
+    set(write_block_size 4)
+    message(WARNING "slot0_partition write block size devicetree parameter is missing, assuming write block size is 4")
+  endif()
```

Comment:
IMHO this should not be done this way.
This should be MCUboot Kconfig option that is either set by SoC configuration Kconfg and the SoC Kconfig may pick that from DTS controller, eventually.
Directly poking at DTS here crosses through module boundaries and entangles, directly, imgtool invocation with DTS definition of Flash partition/controller, and that definitions should belong to driver only.

Reply:
I know there is precedence for that, and I do not like it either. MCUboot is poking at something that does not belong to it, that is hardware property for a driver not a software stack above it.
This injects property of a subsystem into a driver and makes it impossible to independently modify driver without affecting system.
This is one of Flash problems in Zephyr that it is not accessed via common interface but rather through directly poking at internals, and gets impossible to move forward because patterns are repeated and suddenly every device has to have write-block-size or erase-block-size, because other subsystems expect it, even though these can vary at run-time (like in STM32 device, where write block size may change with voltage level applied to device).
Alignment here is a software property, not a hardware property. You can set here N * write-block-size (N is int > 1)) of a device and it will work; the hardware property is here constrain for software.